[A팀] 웹 프론트엔드 파트 코드리뷰용 PR

inaemon opened the pull request:
## #️⃣ 연관된 이슈

작업한 내용과 관련된 **Issue Number**를 작성해주세요.

ex) #이슈번호

## ✅ PR 유형

어떤 **변경 사항**이 있었나요?

- [ ] 새로운 기능 추가
- [ ] 디자인 추가 및 수정
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## 💻 작업 내용

이번 PR에서 작업한 내용에 대해 설명해주세요.

#### [컴포넌트명]

- 내용 1
- 내용 2
- 🖼️ 스크린샷

## 🙏 리뷰 요구사항

리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요.

ex) OO컴포넌트의 OO메소드명 검토 부탁드립니다.

## ✅ 체크리스트(PR 올리기 전 아래 내용을 확인해 주세요)

- [ ] Reviewers를 지정해주세요
- [ ] Assignees는 본인을 선택해주세요
- [ ] label을 선택해주세요

hyesungoh commented on src/assets/webp/back_left_button_gray.webp:
asset들을 public 폴더에 넣어 같은 서버에서 내려줄 수 있을 것 같은데 이렇게 하신 이유가 궁금해요

hyesungoh commented on src/components/button/ChatBot.tsx:
```diff
@@ -0,0 +1,23 @@
+import Image from "next/image";
+import chatbotIcon from "../../assets/svg/chatbot_button.svg";
+
+const FloatingChatbotButton: React.FC = () => {
+  const router = useRouter();
+
+  const handleClick = () => {
+    router.push("/Chatbot/Chatbot");
+  };
+
+  return (
+    <button
+      onClick={handleClick}
+      className="absolute bottom-4 right-4 flex items-center justify-center transition duration-300 ease-in-out"
+    >
+      <Image src={chatbotIcon} alt="히트존 챗봇" width={48} height={48} />
+    </button>
+  );
+};
```
공통 컴포넌트치고는 로직과 UI가 많이 추상화되어 있는 거 같아요

hyesungoh commented on src/components/button/ChatBot.tsx:
```diff
@@ -0,0 +1,23 @@
+import Image from "next/image";
+import chatbotIcon from "../../assets/svg/chatbot_button.svg";
+
+const FloatingChatbotButton: React.FC = () => {
```
`FC` 타입을 사용하신 이유가 궁금해요.

hyesungoh commented on src/components/layout/NavBar.tsx:
```diff
@@ -0,0 +1,51 @@
+  ];
+
+  return (
+    <div className="fixed bottom-0 w-full max-w-[500px] bg-grayscale-0 border-t border-grayscale-10 flex justify-around p-2">
```
`nav`와 같은 시멘틱 태그를 이용할 수 있지 않을까요?

이용한다면 어떤 이점이 있을까요?

hyesungoh commented on src/pages/Culture/Culture.tsx:
일반적으로 url에서는 대소문자 구분이 없긴 합니다. (구분되는 환경도 있지만요)
각 페이지는 소문자로 시작하면 어떨까요?

hyesungoh commented on src/pages/Main/components/Dropdown.tsx:
```diff
@@ -0,0 +1,54 @@
+import Image from "next/image";
+import stadiumDropdownIcon from "../../../assets/webp/stadium_dropdown.webp";
+
+interface DropdownProps {
+  options: string[];
+  selectedOption: string;
+  onSelect: (option: string) => void;
+}
+
+export default function Dropdown({ options, selectedOption, onSelect }: DropdownProps) {
```
서면 리뷰로도 남긴 내용인데

Next.js config의 pageExtension 설정을 안하고 이렇게 두면

`/Main/components/Dropdown`도 페이지로 잡히게 돼요

hyesungoh commented on src/pages/_app.tsx:
```diff
@@ -0,0 +1,65 @@
+
+
+const MyApp: React.FC<AppProps> = ({ Component, pageProps: { session, ...pageProps } }) => {
+  // 현재 URL 경로 가져오기
+  const [currentPath, setCurrentPath] = useState<string>('');
+
+  useEffect(() => {
+    // 클라이언트 측에서 경로 설정
+    setCurrentPath(window.location.pathname);
+  }, []);
+
+
+  // 현재 경로에 따라 컴포넌트 선택
+  const renderComponent = () => {
+    switch (currentPath) {
+        case '/login':
+            return <Onboarding />;
+        case '/signup':
+            return <SignupPage />;
+        case '/':
+            return <Main />;
+        case '/recommend/question':
+            return <Question />;
+        case '/guide':
+            return <Guide />;
+        case '/culture':
+            return <Culture />;
+        case '/mypage':
+            return <MyPage />;
+    }
+  };
```
이렇게 클라이언트 사이드에 라우팅을 전부 위임하게 되면 Next.js를 사용하는 이유가 없어지는 것 아닌가요?

이렇게 하신 이유는 소문자를 대응하기 위함인가요?

inaemon replied:
꼼꼼하게 봐주셔서 감사합니다!
레포지토리를 체계젹으로 폴더화하여 정리하려다보니 url과 맞추기가 힘들었던 것 같습니다ㅠ
예를 들어, 메인홈 페이지는 main 폴더에서 관리하고 싶은데 /main을 경로로 하기는 싫어서 위와 같이 작성하였는데 더 좋은 방법이 있을까요?!

hyesungoh replied:
현재 작성하신 모습은 굉장한 안티패턴이라고 생각돼요

라우팅을 제공하는 프레임워크에서 라우팅을 구현한 것이라서요
> 성능적으로도 그렇고요

말씀하신것처럼 관리 부분을 나누고 싶다면 페이지 단에서 굉장히 큰 페이지라고 부를만한 컴포넌트를 사용하시면 되긴 하겠지만, 그것도 권장되는 방향은 아니라고 생각해요.

```
ㄴ pages
    ㄴ index.tsx
ㄴ components
  ㄴ main
      ㄴ Foo.tsx
```

이런식 혹은 다른 디렉터리 명을 써서 구분하면 될 거 같은데, 다른 많은 오픈소스 프로젝트나 예시를 찾아보시면 좋을 거 같아요

inaemon replied:
꼼꼼한 피드백 정말 감사합니다! 당시에 바빴는데 지금 바로 리팩토링하도록 하겠습니다 ㅎㅎ

inaemon committed:
(#49) 🚀 feat: KT 이미지 이벤트 로직 수정, 메인홈에서 스타디움 컴포넌트 별로 화면 너비 반응형으로 전달
메인홈 KT 컴포넌트에서 좌석 잘 눌리게 오차값 수정
메인홈 KT 컴포넌트에서 클릭한 좌석에서는 디폴트 좌석으로만 넘어가도록 수정
메인홈에서 잠실, KT 컴포넌트에 화면 너비값을 props로 전달하도록 로직 수정

inaemon committed:
(#49) ✨ style: 반응형 관련 더미 데이터 생성

inaemon committed:
(#49) 🚀 feat: 너비에 따른 잠실, KT 이미지 크기 조정 이벤트 구현

inaemon committed:
(#49) 🎨 design: 메인홈 잠실 이미지 크기 반응형 구현

inaemon committed:
(#49) 🎨 design: 메인홈 KT 이미지 크기 반응형 구현

inaemon committed:
(#50) 🐻 INAE: 메인홈 페이지 야구장 좌석 반응형 구현 및 머지
Merge pull request #50 from KUSITMS-30th-TEAM-A/design/#49-reaction
(#49) design: 메인홈 반응형 구현

inaemon committed:
(#51) 🐞 fix: 이미지 최대 너비 height -> width로 수정

inaemon committed:
(#51) 🐞 fix: 메인홈 페이지 잠실 이미지 반응형 웹 크기 추가

inaemon committed:
(#51) 🐞 fix: 메인홈 페이지 KT 이미지 반응형 웹 크기 추가

inaemon committed:
(#51) 🐞 fix: 메인홈 야구장 컴포넌트 일반 폰 기종에서 좌석 인식율을 높이기 위해 이미지 크기 더 작게 조정

inaemon committed:
(#51) 🐞 fix: KT 야구장 인식율 낮은 좌석 색상 오차범위 더 넓게 수정

inaemon committed:
(#51) 🐞 fix: KT 야구장 터치 이벤트 수정

inaemon committed:
(#53) 🐻 INAE: 메인홈 페이지 야구장 이미지 웹 반응형 추가 및 머지

(#51) fix: 메인홈 웹 반응형 구현
Merge pull request #53 from KUSITMS-30th-TEAM-A/fix/#51-mainWebReaction

2unhi committed:
(#48) 🎨 design: 잠실종합운동장 이미지 수정 및 익사이팅존 색상값 수정

2unhi committed:
(#48) 🎨 design: 수원KT위즈파크 이미지 수정 및 마진 변경

2unhi committed:
(#48) 🎨 design: 구역가이드 첫 페이지 배경 색상 수정 및 드롭다운 박스 흰색 배경 추가

2unhi committed:
(#48) 🎨 design: 구역가이드 상세 페이지 디자인 수정 및 좌석 드롭다운 박스 흰색 배경 추가

2unhi committed:
(#54) 🐰 EUNHAK: 구역 가이드 디자인, 메인홈 야구장 이미지 수정 및 머지

수정된 구역 가이드 디자인을 적용 및 메인홈 야구장 이미지 수정

inaemon committed:
(#52) 🐞 fix: KT 야구장 이미지 일반 폰 기종 최적 크기로 조정

inaemon committed:
(#52) 🐞 fix: 야구장 이미지 미니 폰 기종 크기 조정

inaemon committed:
(#52) 🐞 fix: 야구장 이미지 프로 폰 기종 최적 크기로 조정

inaemon committed:
(#52) 🐞 fix: 잠실, KT 야구장 잘 안 눌리는 좌석 오차 범위 조정

inaemon committed:
(#52) 🐞 fix: KT 야구장 로직 복구 (좌석 클릭시 딜레이 현상 해결)

inaemon committed:
(#52) 🎨 design: 메인홈 배경색 수정

inaemon committed:
(#52) 🎨 design: 구역 추천 질문 1 페이지 배경 일러스트가 텍스트 가림 현상 수정

inaemon committed:
(#52) 🚀 feat: 반응형 이벤트를 다루는 커스텀 훅 생성

inaemon committed:
(#52) 🚀 feat: 구역 추천 질문 헤더 반응형 적용

inaemon committed:
(#52) 🚀 feat: 반응형 커스텀 훅 반응형 데이터와 연동

inaemon committed:
(#52) 🚀 feat: 추천 결과 페이지 폰 기종별 반응형 적용

inaemon committed:
(#52) 🎨 design: 구역 추천 결과 페이지 프로필 이미지 백그라운드 프론트에서 처리
미니폰 기종에서 프로필 이미지 화질 이슈로 인해 백엔드에서 백그라운드를 제외한 프로필 이미지만 주는 것으로 수정

inaemon committed:
(#52) 🎨 design: 구역 추천 결과 페이지 추천 구역 설명 텍스트 넘침 현상 해결

inaemon committed:
Merge branch 'develop' into design/#52-recommendedResultReaction

inaemon committed:
(#56) 🐻 INAE: 반응형 추천 결과 페이지 구현 및 머지

(#52) 🎨 design: 추천 결과 페이지 반응형 구현
Merge pull request #56 from KUSITMS-30th-TEAM-A/design/#52-recommendedResultReaction

inaemon committed:
🐞 fix: 커밋 충돌 해결

2unhi committed:
(#55) 🎨 design: 메인 홈 코치마크 구현

inaemon committed:
(#57) 🎨 design: 잠실-프로폰, KT-일반폰 반응형 크기 최적화

inaemon committed:
(#57) 🐞 fix: 잠실 익사이팅석 색상 이벤트값 노랑색으로 변경

inaemon committed:
(#57) 🚀 feat: KT 색상 오차값 수정

inaemon committed:
(#57) 🎨 design: 추천 질문 1 페이지 헤더바 미니폰 반응형 추가

inaemon committed:
(#57) 🚀 feat: 구역 추천 결과 페이지에서 자세히 보러가기 버튼 클릭시 좌석 리스트 API 호출하여 드롭다운 활성화

inaemon committed:
(#57) 🚀 feat: 외부 클릭 감지 커스텀 훅 생성

inaemon committed:
(#57) 🚀 feat: 메인홈, 구역가이드 페이지 드롭다운에 외부 영역 클릭시 닫히는 이벤트 적용

inaemon committed:
(#57) 🚀 feat: 구역가이드 상세 페이지 드롭다운에 외부 영역 클릭시 닫히는 이벤트 적용

2unhi committed:
(#55) 🚀 feat: 헤더 책 모양 클릭 시 코치마크 이동 및 닫기 이벤트 구현

inaemon committed:
(#58) 🐻 INAE: 전체 페이지 세부 디자인 수정과 반응형 최적화 및 머지

(#57) 🎨 design: 세부 디자인 수정 및 반응형 최적화
Merge pull request #58 from KUSITMS-30th-TEAM-A/design/#57-designModification

inaemon committed:
Merge branch 'develop' into design/#55-coachmark

2unhi committed:
(#59) 🐰 EUNHAK: 메인 홈 코치마크 구현 및 머지

메인 홈에서 보여지는 코치마크를 디자인하고 필요한 이벤트 구현

inaemon committed:
(#60) 🎨 design: 코치마크 아이콘 클릭시에만 띄우도록 수정

inaemon committed:
(#60) 🎨 design: 메인홈 배경색 수정

inaemon committed:
(#60) 🎨 design: 메인홈 야구장 3루 배경색 수정

inaemon committed:
(#60) 🎨 design: 야구 직관 TIP 이미지 딜레이 최적화

inaemon committed:
(#60) 🎨 design: favicon 설정

inaemon committed:
(#60) 🎨 design: 브라우저 타이틀 설정

inaemon committed:
(#61) 🐻 INAE: 야구 직관 Tip 이미지 딜레이 현상 조정 외 세부 디자인 수정 및 머지

(#60) 🎨 design: 야구 직관 Tip 이미지 딜레이 현상 조정 외 세부 디자인 수정
Merge pull request #61 from KUSITMS-30th-TEAM-A/design/#60-imageDelay

inaemon committed:
(#63) 🎨 design: 메인홈 스크롤 어필 컴포넌트 생성

inaemon committed:
(#63) 🎨 design: 메인홈 하단에 스크롤 어필 컴포넌트 출력

inaemon committed:
(#63) 🎨 design: 메인홈 하단에 스크롤 어필 컴포넌트 디자인 수정

2unhi committed:
(#62) 🎨 design: 코치마크 3단계 이미지 수정

2unhi committed:
(#62) 🎨 design: 야구직관 Tip 이미지 교체

inaemon committed:
(#63) 🎨 design: 메인홈 스티키 현상 해결

2unhi committed:
(#62) 🎨 design: 준비중인 서비스 디자인 및 설문조사 URL 연결

2unhi committed:
(#62) 🎨 design: 구역 추천 결과 페이지 Tip 버튼 수정

2unhi committed:
(#62) 🎨 design: 추천 다시 받기 버튼 배경 색상 및 스트로크 추가

2unhi committed:
(#62) 🎨 design: 구역 가이드 Tip 버튼 수정

2unhi committed:
(#64) 🐰 EUNHAK: 코치마크, Tip 버튼, 준비중인 서비스 페이지 작업 및 머지

수정된 코치마크, Tip 버튼 디자인 작업 및 준비중인 서비스 페이지 구현

inaemon committed:
(#63) 🚀 feat: 스크롤 감지 이벤트 커스텀 훅 제작

inaemon committed:
(#63) 🚀 feat: 참조를 걸어둔 태그에서의 스크롤을 감지하는 커스텀 훅 제작

inaemon committed:
(#63) 🚀 feat: 메인홈에 스크롤 이벤트 장착

inaemon committed:
(#63) 🚀 feat: 스크롤 내림 감지시 컴포넌트를 숨기는 커스텀 훅 제작

inaemon committed:
(#63) 🚀 feat: 메인홈 첫 방문시 스크롤 어필 컴포넌트 띄우도록 설정

inaemon committed:
(#63) 🎨 design: 스크롤 어필 컴포넌트의 부모 컴포넌트를 구역 찾으러가기 버튼으로 셋팅

inaemon committed:
(#63) 🚀 feat: 모바일에서도 스크롤 감지가 동작하도록 useRefScroll에 터치 이벤트 추가

inaemon committed:
(#63) 🎨 design: 스크롤 어필 컴포넌트를 부모 컴포넌트 기준이 아닌 현재 보이는 화면을 기준으로 렌더링하도록 반응형 효과 적용

inaemon committed:
(#63) ✨ style: useVisibilityWithFade 훅 이벤트 별로 주석 추가

inaemon committed:
(#63) 🚀 feat: MS Clarity 설정

inaemon committed:
(#66) 🐻 INAE: MS Clarity 설정 외 메인홈 스크롤 효과 적용 및 머지

(#63) 🎨 design:  MS Clarity 설정 외 메인홈 스크롤 효과
Merge pull request #66 from KUSITMS-30th-TEAM-A/design/#63-ms

inaemon committed:
(#47) 🚀 feat: 날짜 커스텀 훅 useDate 구현

inaemon committed:
(#47) 🔄 refactor: 채팅창 날짜 배너 컴포넌트화

inaemon committed:
(#47) 🔄 refactor: 채팅창 말풍선 3종류 컴포넌트화

inaemon committed:
(#47) 🔄 refactor: 채팅창 루키 프로필 컴포넌트화

inaemon committed:
(#47) 🔄 refactor: 메시지 말풍선 컴포넌트명 수정

2unhi committed:
(#65) 🚀 feat: 카카오톡 공유 기본 기능 구현

2unhi committed:
(#65) 🚀 feat: 카카오톡 공유하기 유형 닉네임, 해시태그 연동

2unhi committed:
(#65) 🔄 refactor: 메시지 커스텀 파일 분리

inaemon committed:
(#47) ✨ style: 챗봇 페이지 프론트 더미 데이터 정리

inaemon committed:
(#47) 🎨 design: 챗봇 페이지 복구

inaemon committed:
(#47) 🚀 feat: 야구장 페이지 전환시 딜레이 효과 제거

inaemon committed:
(#47) 🎨 design: 챗봇 플로팅 버튼 디자인 변경

2unhi committed:
(#65) 🚀 feat: 카카오톡 공유하기 템플릿 적용 및 결과 연동

2unhi committed:
(#67) 🐰 EUNHAK: 추천 결과 카카오톡 공유하기 기능 구현 및 머지

구역 추천 결과 페이지 우측 상단에 카카오톡 공유하기 기능 구현

inaemon committed:
(#47) 🔄 refactor: 챗봇 FAQ 카테고리바 더미 데이터 리팩토링

inaemon committed:
(#47) 🎨 design: 챗봇 FAQ 카테고리바 호버 색상과 커서 포인트 추가 및 디자인 수정

inaemon committed:
(#47) 🎨 design: 챗봇 FAQ 카테고리바를 닫은 경우에 보이는 컴포넌트 생성

inaemon committed:
(#47) 🎨 design: 챗봇 FAQ 카테고리바에 닫는 핑크바 버튼 추가 및 디자인 수정

2unhi committed:
(#68) 🎨 design: 메인 홈 야구직관 Tip 버튼 수정

inaemon committed:
(#47) 🚀 feat: 챗봇 FAQ 카테고리 토글 이벤트 설정

2unhi committed:
(#68) 🎨 design: 메인 홈 스크롤 시간 1.5초로 수정

inaemon committed:
(#47) 🎨 design: 챗봇 채팅 필드 디자인 변경

inaemon committed:
(#47) 🎨 design: 핑크 버튼 호버 색상 추가

inaemon committed:
(#47) 🎨 design: 챗봇 프로필 정사각형 크기 변경

2unhi committed:
(#68) 🎨 design: 구역 추천 결과 페이지 색상 수정

inaemon committed:
(#47) ✨ style: 말풍선 컴포넌트명 수정

2unhi committed:
(#68) 🎨 design: 구역 추천 페이지에서 야구문화 연결 버튼 추가

inaemon committed:
(#47) 🎨 design: 챗봇 첫 페이지 디자인 수정

2unhi committed:
(#68) ✨ style: 가이드 관련 패딩 값 조정

inaemon committed:
(#47) 🎨 design: 챗봇 루키 채팅 구성 세트 컴포넌트화

inaemon committed:
(#47) 🚀 feat: 스타디움을 선택한 경우에만 FAQ 카테고리 렌더링

2unhi committed:
(#68) 🚀 feat: 구역 가이드에서 좌석 클릭 시 헤더에 각 야구장 명칭이 나오도록 수정

inaemon committed:
(#47) 🎨 design: 챗봇 스타디움 선택 카테고리 제공하지 않는 구장은 비활성화 디자인으로 수정

inaemon committed:
(#47) 🚀 feat: 챗봇 스타디움 선택 카테고리 제공하지 않는 구장은 이벤트 비활성화

inaemon committed:
(#47) 🚀 feat: 챗봇 스타디움 구장 선택시 카테고리 재선택 비활성화

inaemon committed:
(#47) 🎨 design: 챗봇 높이 끝까지 스크롤 가능하도록 디자인 수정

inaemon committed:
(#47) 🎨 design: 유저 채팅창 컴포넌트화

inaemon committed:
(#47) 🎨 design: 챗봇 구장 선택시 챗봇 답변챗 출력

inaemon committed:
(#47) 🐞 fix: 루키 채팅 컴포넌트 처음챗 내용 올바른 값으로 수정

inaemon committed:
(#47) 🚀 feat: 챗봇 페이지 대화 길어지면 자동으로 맨 아래로 스크롤되는 이벤트 적용

inaemon committed:
(#47) 🐞 fix: 챗봇 페이지 맨 아래 내용까지 보이도록 패딩 추가

inaemon committed:
(#47) 🚀 feat: 챗봇 FAQ 카테고리 클릭시 발동 이벤트 추가 및 연동

inaemon committed:
(#47) 🎨 design: 챗봇 FAQ 카테고리 데이터 컴포넌트화하여 관리

inaemon committed:
(#47) 🎨 design: 챗봇 FAQ 카테고리 백엔드 연동이 필요한 세부 컴포 디자인 적용

inaemon committed:
(#47) 🎨 design: 챗봇 FAQ 카테고리 백엔드 연동이 필요한 세부 컴포 디자인 적용

2unhi committed:
(#68) 🎨 design: 구역 추천 결과 Tip 버튼 수정

inaemon committed:
(#47) 🎨 design: 모든 헤더바 맨 앞에 렌더링되도록 z-20 적용

inaemon committed:
(#47) 🎨 design: 챗봇 세부 카테고리 아이콘 데이터 추가

2unhi committed:
(#68) 🔄 refactor: 준비중인 서비스 컴포넌트화

2unhi committed:
(#69) 🐰 EUNHAK: 수정된 디자인 적용, 구장별 헤더 변경, 준비중인 서비스 컴포넌트화 및 머지

QA로 반영한 수정된 디자인 적용, 구장별 구역가이드 헤더 변경, 준비중인 서비스 컴포넌트화

inaemon committed:
(#47) 🚀 feat: 챗봇 가이드 API 핸들러 생성

inaemon committed:
(#47) 🎨 design: 챗봇챗 컴포넌트 FAQ 종류별 커스텀 디자인 디테일 수정

inaemon committed:
(#47) 🎨 design: 챗봇 사용설명서 이미지 추가 및 챗봇챗 이미지 디자인 수정

inaemon committed:
(#47) 🚀 feat: 챗봇 가이드 API 답변 부모 컴포넌트로 전달하도록 수정

inaemon committed:
(#47) 🎨 design: 채팅창에 챗봇 가이드 API 답변받으면 루키 채팅 컴포에 출력

inaemon committed:
(#47) 🎨 design: 사용자가 선택한 세부 가이드 카테고리와 매핑되는 답변만 출력되도록 변경

inaemon committed:
(#47) 🎨 design: 사용자가 선택한 세부 가이드 카테고리와 매핑되는 답변 여러 개 선택시 모두 출력

inaemon committed:
(#47) 🎨 design: 가이드 답변 받으면 채팅방 맨 아래로 자동 스크롤 기능 활성화화

inaemon committed:
(#47) 🔄 refactor: 가이드 답변 저장 타입 변경

inaemon committed:
(#47) 🎨 design: 서브 카테고리 선택시 사용자 말풍선 출력

inaemon committed:
(#47) 🎨 design: 챗봇 채팅창 카테고리 컴포넌트 아래에 서브 카테고리 답변 배치

inaemon committed:
(#47) 🎨 design: 챗봇 채팅창 세부 카테고리 답변 컴포넌트 디자인 수정

inaemon committed:
Merge branch 'develop' into design/#47-chatbotPage

inaemon committed:
(#71) 🐻 INAE: 챗봇 가이드 구현 및 머지

(#47) 🎨 design:  챗봇 가이드 구현
Merge pull request #71 from KUSITMS-30th-TEAM-A/design/#47-chatbotPage

inaemon committed:
🚀 feat: MS Clarity 스크립트 수정

inaemon committed:
🐞 fix: ESLint 오류 해결 (img 태그를 Image 태그로 변경)

inaemon committed:
(#72) 🐞 fix: 챗봇 가이드 선택한 stadium 값을 백엔드 통신용 파라미터 데이터로 변환하여 API 요청

inaemon committed:
(#72) 🚀 feat: 챗봇 가이드 API 받은 데이터 타입 변환 및 파싱하여 반환

inaemon committed:
(#72) ✨ style: 챗봇 데이터 관련 함수에서 변수명 변경 및 주석 추가

inaemon committed:
(#72) 🚀 feat: 백엔드로부터 받은 answer 데이터의 타입을 string array에서 string으로 처리하도록 수정

inaemon committed:
(#72) 🎨 design: new line 반영 말풍선 컴포넌트 제작

2unhi committed:
(#70) 🎨 design: 야구 문화 페이지 상단 탭바, 구장 내부/외부 컴포넌트화 및 기본 디자인 작업

2unhi committed:
(#70) 🎨 design: 각 박스 클릭 시 세부 내용을 확인하는 모달창

inaemon committed:
(#72) 🎨 design: 루키 new line 반영 말풍선 pre tag 대신에 css 방식으로 변경

inaemon committed:
(#72) 🎨 design: 루키 new line 반영 꼬랑지 말풍선 제작

inaemon committed:
(#72) 🎨 design: 루키 화이트 리스트 말풍선 제작

inaemon committed:
(#72) 🎨 design: 루키 화이트 리스트 꼬랑지 말풍선 제작

inaemon committed:
(#72) 🎨 design: 루키 이미지 말풍선 제작

inaemon committed:
(#72) 🔄 refactor: 카테고리챗 컴포넌트화한 말풍선으로 출력

2unhi committed:
(#74) 🎨 design: 코치마크 2번째 단계 수정

2unhi committed:
(#74) 🚀 feat: 헤더 우측 알림 버튼 클릭 시 준비중인 페이지로 이동

2unhi committed:
(#74) 🎨 design: 추천 결과 구역 설명 개행 및 양쪽 정렬

inaemon committed:
(#72) ✨ style: 주석 추가 및 코드 안정성을 위해 세미콜론 추가

inaemon committed:
(#72) 🎨 design: 루키 채팅 셋 컴포넌트에 줄바꿈 인식하는 말풍선 타입 추가

inaemon committed:
(#72) 🎨 design: 루키 채팅 셋 컴포넌트에 이미지 말풍선 타입 추가

inaemon committed:
(#72) ✨ style: 루키 채팅 셋 컴포넌트에 주석 설명 추가

inaemon committed:
(#72) 🎨 design: 가이드 답변은 줄바꿈 꼬랑지 말풍선에 출력하도록 수정

inaemon committed:
(#72) 🐞 fix: 줄바꿈 꼬랑지 말풍선 컴포넌트에서 꼬랑지 안 나오는 현상 해결

inaemon committed:
(#72) 🎨 design: 초기에 챗봇 가이드 이미지도 띄우도록 설정

inaemon committed:
(#72) 🎨 design: 챗봇 말풍선 너비 일반화 및 모듈화

2unhi committed:
(#75) 🐰 EUNHAK: 코치마크, 추천 결과 구역 설명 디자인 수정 및 알림창 클릭 시 준비중인 서비스 페이지로 이동 및 머지

코치마크, 추천 결과 중 구역 설명에 대한 디자인 수정 및 헤더 우측 알림창 클릭 시 준비중인 서비스 페이지로 이동

inaemon committed:
(#72) 🎨 design: 채팅창 이미지 클릭시 모달창으로 띄우기

inaemon committed:
(#72) 🚀 feat: 이미지 모달 커스텀 훅 제작

inaemon committed:
(#72) 🚀 feat: 챗봇 가이드 FAQ 카테고리바 외부 영역 클릭시 닫히도록 이벤트 구현

inaemon committed:
(#72) 🚀 feat: 챗봇 가이드 FAQ 카테고리 선택시 카테고리바 닫히는 이벤트 추가

2unhi committed:
(#76) 🚀 feat: 추천 결과 페이지 예매 버튼에서 해당 홈 구장을 가장 위로 배치

2unhi committed:
(#76) 🎨 desgin: 코치마크 3단계에서 홈으로 이동하는 버튼 텍스트 수정

inaemon committed:
(#72) 🚀 feat: 챗봇 채팅창 영역 뿐만 아니라 전체 윈도우 영역도 스크롤 아래로 내려가도록 자동 스크롤 이벤트 수정

inaemon committed:
(#72) 🚀 feat: 챗봇 페이지 처음 접속시에는 전체 윈도우 자동 스크롤 현상 막기

inaemon committed:
(#72) 🎨 design: 카테고리바 원래 위치로 복원

2unhi committed:
(#76) 🎨 design: Header 고정 및 챗봇 헤더 위치 수정

2unhi committed:
(#76) 🎨 design: 준비중인 서비스 페이지 이미지 및 워딩 수정

inaemon committed:
(#72) 🎨 design: 백엔드로부터 전달받은 이미지 메시지도 출력되도록 수정

2unhi committed:
(#76) 🔄 refactor: 카카오톡 공유 SDK 관련 코드 수정

inaemon committed:
(#72) 🔄 refactor: 루키 챗 내용물 종류별로 커스텀 말풍선 출력되도록 구조 정규화

inaemon committed:
(#72) 🐞 fix: 반복 렌더링시 고유 key값을 상위 div 태그에 할당

inaemon committed:
(#72) 🎨 design: 챗봇 채팅창 아래 여백 줄임

2unhi committed:
(#76) 🎨 design: 드롭다운 박스 아이콘 수정

2unhi committed:
(#76) 🎨 design: 구역 선택 드롭박스 다운 애니메이션 적용

inaemon committed:
(#72) 🎨 design: 이동 버튼이 달린 루키 채팅 말풍선 컴포넌트 제작

2unhi committed:
(#76) 🚀 feat: 모달창 스크롤 방지 Hook 추가 및 적용

inaemon committed:
(#72) 🚀 feat: 챗봇 가이드 API 데이터 link, linkName 추가 및 연동

2unhi committed:
(#77) 🐰 EUNHAK: 카카오톡 공유 SDK, 예매 다이얼로그 홈/원정 순서, QA 디자인 수정 및 머지

카카오톡 공유 SDK 관련 검토, 예매 다이얼로그 홈/원정 순서 조정, QA로 수정된 디자인 전체 적용

inaemon committed:
(#72) 🚀 feat: 챗봇 가이드 link, linkName 데이터 모든 API 로직 및 디자인 연동

inaemon committed:
(#78) 🐻 INAE: 챗봇 가이드 세부 디테일 완성 및 머지

(#72) 🚀 feat: 챗봇 가이드 세부 디테일 수정 및 API 데이터 추가
Merge pull request #78 from KUSITMS-30th-TEAM-A/feat/#72-chatbotGuide

inaemon committed:
(#79) 🚀 feat: 챗봇 클로바 이벤트 핸들러 구현 및 input field에 연동

inaemon committed:
(#79) 🐞 fix: 챗봇 클로바 API 연동 로직 올바르게 수정

inaemon committed:
(#79) 🎨 design: 챗봇 입력창 전체가 눌리도록 너비 꽉 차게 수정

inaemon committed:
(#79) 🎨 design: 챗봇 텍스트 전송 후 입력창에 입력값 초기화

inaemon committed:
(#79) 🎨 design: 챗봇 클로바 입력시 로딩 상태 디자인 구현

inaemon committed:
(#79) 🐞 fix: 이미지 메시지 말풍선 오타 수정

inaemon committed:
(#79) 🎨 design: 챗봇 루키 프로필 이미지 네모 상자 탈출 현상 해결 (이미지 크기 수정)

inaemon committed:
(#79) 🔄 refactor: 챗 컴포넌트는 하나의 디렉토리 안으로 정리

inaemon committed:
(#79) 🎨 design: 채팅 로딩시 로딩 이미지 아래에 설명 문구 추가

inaemon committed:
(#79) 🎨 design: 채팅 이미지 모달창 닫기 버튼 아이콘을 이미지로 수정

inaemon committed:
(#79) 🔄 refactor: 시간 순대로 챗 컴포넌트 출력되도록 리팩토링

inaemon committed:
(#79) 🎨 design: 챗봇 클로바 답변 렌더링

inaemon committed:
(#79) 🚀 feat: 챗봇 클로바 답변 기다리는 로딩 상태일 때도 스크롤 자동으로 내리는 기능 활성화

inaemon committed:
(#79) 🚀 feat: 엔터키 입력시 챗봇 클로바 전송 버튼 눌리는 이벤트 추가

inaemon committed:
(#79) 🐞 fix: 이미지 관련 컴포넌트 임포트

inaemon committed:
(#79) 🐞 fix: 사용하지 않는 컴포넌트 임포트 삭제

inaemon committed:
(#79) 🐞 fix: 배포 오류 (문법 오류) 해결

inaemon committed:
(#80) 🐻 INAE: 챗봇 클로바 기능과 시간순 렌더링 구현 및 머지

(#79) 🚀 feat: 챗봇 클로바 기능 구현 및 챗봇 가이드 디테일 수정
Merge pull request #80 from KUSITMS-30th-TEAM-A/feat/#79-chatbotClova

inaemon committed:
(#73) 🐞 fix: 챗봇 관련 이미지 챗 컴포와 이미지 모달창 임시 너비 높이 설정

inaemon committed:
(#73) 🎨 design: 시작 페이지 배경 색상 변경: 연분홍 -> 화이트

inaemon committed:
(#73) 🎨 design: 온보딩 마지막 페이지 디자인 수정

inaemon committed:
(#73) 🎨 design: 온보딩 마지막 페이지 문구 수정

inaemon committed:
(#73) 🎨 design: 온보딩 마지막 페이지 헤더바 변경

2unhi committed:
(#70) 🎨 design: 야구 문화 전체보기 클릭 시 나오는 해당 리스트 디자인

inaemon committed:
(#73) 🚀 feat: 홈화면 온보딩 렌더링 로직 구현

inaemon committed:
(#73) 🚀 feat: 온보딩 렌더링 이벤트 연동

inaemon committed:
(#73) 🎨 design: 온보딩 마지막 페이지 배경 이미지 변경 및 루키 위치 조정

inaemon committed:
(#73) 🐞 fix: 메인 페이지 스타디움 변수 훅으로 관리하도록 명청 변경

2unhi committed:
(#70) 🎨 design: 야구 문화 Header 디자인 수정

inaemon committed:
(#73) 🚀 feat: 새 탭으로 들어온 경우만 온보딩을 진행하고 그 이후엔 메인 페이지를 렌더링하도록 이벤트 구현

2unhi committed:
(#70) 🚀 feat: 야구 문화 구장별 페이지 렌더링

inaemon committed:
(#81) 🐻 INAE: 온보딩 플로우 구현 및 머지

(#73) 🎨 design: 온보딩 플로우 구현
Merge pull request #81 from KUSITMS-30th-TEAM-A/design/#73-onboardingFlow

inaemon committed:
🐞 fix: 배포 오류 해결 (문법 오류 해결)

2unhi committed:
(#70) 🎨 design: 전체보기 버튼 통일 및 더미데이터 추가

inaemon committed:
(#82) 🎨 design: 온보딩 마지막 페이지 루키 위에 문구 뜨도록 렌더링 순서 수정

2unhi committed:
(#70) 🎨 design: 야구 문화 모달창 수정

inaemon committed:
(#82) ✨ style: 전역 스크롤 훅 컴포넌트명 변경

2unhi committed:
(#70) 🎨 design: 야구 문화 이미지 추가 및 모달창, 박스 크기 조정

2unhi committed:
(#83) 🐰 EUNHAK: 잠실종합운동장 야구문화 페이지 디자인 및 머지

잠실종합운동장 야구문화 페이지 디자인

inaemon committed:
🐞 fix: 배포 오류 해결 (webp -> svg 이미지 확장자 변경)

2unhi committed:
(#84) 🔄 refactor: 야구 문화 메인 화면 리팩토링

2unhi committed:
(#84) 🎨 design: 전체보기 버튼 및 디테일 페이지

inaemon committed:
(#82) 🚀 feat: 마우스 휠 이벤트 추가하여 스크롤바가 안 생기는 경우에도 스크롤 이벤트가 작동하도록 커스텀 스크롤 훅 수정

inaemon committed:
(#82) ✨ style: 사용하지 않는 변수 제거

inaemon committed:
(#82) 🎨 design: 스타디움 전역 상태 동기화를 위한 Provider 및 커스텀 훅 제작

inaemon committed:
(#82) 🚀 feat: useStadiumSelector 훅과 메인홈 컴포넌트에 스타디움 전역 상태 동기화 적용

inaemon committed:
(#82) 🚀 feat: 스타디움 API 호출시 유효성 검사 추가

inaemon committed:
(#82) 🚀 feat: 스타디움 관련 데이터는 모두 전역으로 관리하도록 StadiumContext에 추가

2unhi committed:
(#84) 🎨 design: 야구문화 페이지에서 사용하는 Tab 분리

inaemon committed:
(#82) 🚀 feat: 메인홈과 가이드 페이지 스타디움 값 동기화 로직 구현

inaemon committed:
(#82) 🎨 design: 최상단 div에서 contents가 overflow될 수 있도록 다시 복구

inaemon committed:
(#82) 🎨 design: 챗봇 백엔드에서 받은 이미지를 띄우는 모달창의 최대 너비 지정

inaemon committed:
(#85) 🐻 INAE: 스타디움 전역 상태 관리와 디테일 디자인 수정 및 머지

(#82) 🎨 design:  렌더링 관련 디테일 디자인 수정
Merge pull request #85 from KUSITMS-30th-TEAM-A/design/#82-detailDesignRendering

2unhi committed:
(#86) 🐰 EUNHAK: 야구문화 페이지 컴포넌트화 및 머지

야구문화 페이지에서 확인할 수 있는 먹거리와 즐길거리 카테고리 컴포넌트화

inaemon committed:
(#87) 🎨 design: 메인홈 배경을 높이 full로 채우기

inaemon committed:
(#87) 🚀 feat: 추천 결과에서 문화 페이지로 이동시 스타디움을 쿼리 파라미터로 전달

inaemon committed:
(#87) 🚀 feat: 문화 페이지의 로컬 변수를 추천 결과로부터 받은 쿼리 파라미터 값으로 업데이트하는 로직 구현

inaemon committed:
(#87) 🚀 feat: 문화 페이지도 메인홈과 스타디움 값과 연동 (전역 상태 관리 적용)

inaemon committed:
🐻 INAE: 추천 결과 페이지와 문화 페이지 연결 및 머지

(#87) 🎨 design: 구역 추천 페이지와 야구 문화 페이지 연동
Merge pull request #89 from KUSITMS-30th-TEAM-A/design/#87-cultureLink
